sysdeps/managarm/generic/file.cpp

```diff
@@ -1745,6 +1747,16 @@ int sys_ioctl(int fd, unsigned long request, void *arg, int *result) {
 		return EBADF;
 
 	switch(request) {
+	case FIONBIO: {
+		auto mode = reinterpret_cast<int *>(arg);
+		int flags = fcntl(fd, F_GETFL, 0);
+		if(*mode) {
+		    fcntl(fd, F_SETFL, flags | O_NONBLOCK);
+		}else{
+		    fcntl(fd, F_SETFL, flags & ~O_NONBLOCK);
```

Comment:
We should handle errors from fcntl here (by propagating them to the caller of ioctl). We also should use sys_fcntl directly, to avoid trashing `errno`.